src/__tests__/string.test.ts

```diff
@@ -81,6 +81,8 @@ test("uuid", () => {
   const uuid = z.string().uuid("custom error");
   uuid.parse("9491d710-3185-4e06-bea0-6a2f275345e0");
   uuid.parse("00000000-0000-0000-0000-000000000000");
+  uuid.parse("b3ce60f8-e8b9-40f5-1150-172ede56ff74"); // Variant 0 - RFC 4122: Reserved, NCS backward compatibility
+  uuid.parse("92e76bf9-28b3-4730-cd7f-cb6bc51f8c09"); // Variant 2 - RFC 4122: Reserved, Microsoft Corporation backward compatibility
```

Comment:
Forgive my ignorance (I typically only use v4 UUIDs), but do we care about "variants" here or "versions" as specified in RFC 4122? I think we want to be able to validate v1 - v5 UUIDs regardless of the "variant", but I'm not entirely sure I understand what a variant is supposed to be.

Reply:
Looking a little deeper, I think I just misunderstood your initial requirement here, so if it is indeed the ability to use these variants, and the new RegExp properly handles any _version_ I think this is fine, I just want to get some clarity for myself.

Reply:
No worries, most of the UUIDs nowadays are variant 1 however some old systems still generate variant 0 or 2 UUIDs, but according to RFC 4122, they are all valid UUIDs. I only changed the fourth part of the regex which is for variant, it won't affect the version validation (v1 to v5) on the third part of the regex.